compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java

```diff
@@ -567,7 +567,7 @@ private static BIntersectionType defineImmutableRecordType(Location pos, BRecord
         BRecordTypeSymbol recordSymbol =
                 Symbols.createRecordSymbol(origRecordType.tsymbol.flags | Flags.READONLY,
                         getImmutableTypeName(names,  getSymbolFQN(origRecordType.tsymbol)),
-                        pkgID, null, env.scope.owner, pos, SOURCE);
+                        pkgID, null, env.scope.owner, pos, origRecordType.tsymbol.origin);
```

Comment:
In this method `origRecordType.tsymbol` is used in several places, shall we move that into a separate variable?

Reply:
Updated